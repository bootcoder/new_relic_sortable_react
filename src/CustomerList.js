import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './CustomerList.css'

class CustomerList extends Component {
  render () {
    return (
      <div className='CustomerList'>
        <table>
          <tbody>
            <tr>
              <td>ID</td>
              <td>First Name</td>
              <td>Last Name</td>
              <td>Company</td>
            </tr>
          </tbody>
        </table>
        <table>
          <tbody>
            {this.props.customers.map((cust, idx) => {
              return (
                <tr key={idx}>
                  <td>{cust.id}</td>
                  <td>{cust.first_name}</td>
                  <td>{cust.last_name}</td>
                  <td>{cust.company_name}</td>
                </tr>
              )
            })}
          </tbody>
        </table>
      </div>
    )
  }
}

CustomerList.propTypes = {
  customers: PropTypes.array.isRequired
}

export default CustomerList
