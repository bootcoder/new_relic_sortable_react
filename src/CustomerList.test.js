/* global describe, it, expect, beforeEach, mount */

import CustomerList from './CustomerList'

describe('<CustomerList />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(
      <CustomerList
        customers={[]} />)
  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('has param props', () => {
    expect(wrapper.props().customers).toEqual([])
  })
})
