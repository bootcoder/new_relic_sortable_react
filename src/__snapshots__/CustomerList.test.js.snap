// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerList /> renders without crashing 1`] = `
<div
  className="CustomerList"
>
  <table>
    <tbody>
      <tr>
        <td>
          ID
        </td>
        <td>
          First Name
        </td>
        <td>
          Last Name
        </td>
        <td>
          Company
        </td>
      </tr>
    </tbody>
  </table>
  <table>
    <tbody />
  </table>
</div>
`;
